openshift_scalability/utils.py

```diff
@@ -350,7 +380,51 @@ def single_project(testconfig, projname, globalvars):
         pod_handler(testconfig["pods"], globalvars)
     if "rcs" in testconfig:
         rc_handler(testconfig["rcs"], globalvars)
-
+    if globalvars["autogen"]:
+        num_expected = int(globalvars["podnum"])
+        pods_running = []  
+        while len(pods_running) < num_expected: 
+            pods_running = subprocess.check_output(
+                "oc get pods --namespace=%s --no-headers |"
+                " awk '/1\/1/ && /Running/ {print $1;}'" \
+                % globalvars["namespace"], shell=True).splitlines()
+            time.sleep(5)
+
+        for pod in pods_running:
+            rsync = subprocess.check_output(
+                    "oc rsync --namespace=%s /root/.ssh %s:/root/" \
+                    % (globalvars["namespace"], pod), shell=True)
+
```

Comment:
What if this fails? This is a generic question but subprocess.check_output can fail sometimes, shouldn't we have a revert function, which reverts all the things that is done.